xbmc/utils/ExtensionManager.cpp

```diff
@@ -0,0 +1,142 @@
+ *
+ */
+
+#include "ExtensionManager.h"
+
+#include <string>
+#include <vector>
+
+#include "addons/AddonManager.h"
+#include "addons/binary-addons/BinaryAddonBase.h"
+#include "ServiceBroker.h"
+
+using namespace ADDON;
+
+const std::vector<TYPE> ADDON_TYPES = {
+  ADDON_VFS,
+  ADDON_IMAGEDECODER,
+  ADDON_AUDIODECODER
+};
+
+CExtensionManager::CExtensionManager()
+{
+  m_advancedSettings = g_advancedSettingsRef;
+
+  InitAddonExtensions();
+
+  CServiceBroker::GetAddonMgr().Events().Subscribe(this, &CExtensionManager::OnAddonEvent);
+}
+
+CExtensionManager::~CExtensionManager()
+{
+  CServiceBroker::GetAddonMgr().Events().Unsubscribe(this);
+
+  m_advancedSettings.reset();
+  m_addonExtensions.clear();
+}
+
+std::string CExtensionManager::GetDiscStubExtensions() const
+{
+  return m_advancedSettings->m_discStubExtensions;
+}
+
+std::string CExtensionManager::GetMusicExtensions() const
+{
+  std::string extensions(m_advancedSettings->m_musicExtensions);
+  extensions += '|' + GetAddonExtensions(ADDON_VFS);
+  extensions += '|' + GetAddonExtensions(ADDON_AUDIODECODER);
+
+  return extensions;
+}
+
+std::string CExtensionManager::GetPictureExtensions() const
+{
+  std::string extensions(m_advancedSettings->m_pictureExtensions);
+  extensions += '|' + GetAddonExtensions(ADDON_VFS);
+  extensions += '|' + GetAddonExtensions(ADDON_IMAGEDECODER);
+
+  return extensions;
+}
+
+std::string CExtensionManager::GetSubtitleExtensions() const
+{
+  std::string extensions(m_advancedSettings->m_subtitlesExtensions);
+  extensions += '|' + GetAddonExtensions(ADDON_VFS);
+
+  return extensions;
+}
+
+std::string CExtensionManager::GetVideoExtensions() const
+{
+  std::string extensions(m_advancedSettings->m_videoExtensions);
+  extensions += '|' + GetAddonExtensions(ADDON_VFS);
+
+  return extensions;
+}
+
+std::string CExtensionManager::GetAddonExtensions(const TYPE &type) const
+{
+  auto it = m_addonExtensions.find(type);
+  if (it != m_addonExtensions.end())
+    return it->second;
+
+  return "";
+}
+
+void CExtensionManager::InitAddonExtensions()
+{
+  for (auto const type : ADDON_TYPES)
+  {
+    SetAddonExtensions(type);
+  }
+}
+
+void CExtensionManager::SetAddonExtensions(const TYPE& type)
+{
+  std::vector<std::string> extensions;
+  BinaryAddonBaseList addonInfos;
+  CServiceBroker::GetBinaryAddonManager().GetAddonInfos(addonInfos, true, type);
+  for (const auto& addonInfo : addonInfos)
+  {
+    std::string ext = addonInfo->Type(type)->GetValue(ADDON_VFS == type ? "@extensions" : "@extension").asString();
+    if (!ext.empty())
+      extensions.push_back(ext);
+  }
+
+  m_addonExtensions.insert(make_pair(type, StringUtils::Join(extensions, "|")));
+}
+
+void CExtensionManager::OnAddonEvent(const AddonEvent& event)
+{
+  if (typeid(event) == typeid(AddonEvents::Disabled) ||
+      typeid(event) == typeid(AddonEvents::Unload) ||
+      typeid(event) == typeid(AddonEvents::Enabled) ||
+      typeid(event) == typeid(AddonEvents::Load))
```

Comment:
As mentioned in the other pr: don't use Load/Unload. This is for something else entirely

Reply:
ok removed load/unload